caffeine@patapon.info/extension.js

```diff
@@ -141,43 +275,113 @@ class Caffeine extends PanelMenu.Button {
             this._screen = global.display;
             this._display = this._screen;
         }
-
-        this._icon = new St.Icon({
-            style_class: 'system-status-icon',
+ 
+        // Add indicator label for the timer
+        this._timerLabel = new St.Label({
+            y_expand: true,
+            y_align: Clutter.ActorAlign.CENTER,
         });
-        this._icon.gicon = Gio.icon_new_for_string(`${Me.path}/icons/${DisabledIcon}.svg`);
+        this._timerLabel.visible = false;
+        this.add_child(this._timerLabel);
 
+        // Icons
+        this._icon_actived = Gio.icon_new_for_string(`${Me.path}/icons/${EnabledIcon}.svg`);;
+        this._icon_desactived = Gio.icon_new_for_string(`${Me.path}/icons/${DisabledIcon}.svg`);
+        this._indicator.gicon = this._icon_desactived;
+
+        // Manage night light and allow blank screen
+        this._night_light = false;
+        this._allow_blank= false;
+        
+        /* Inhibited flag value
+        * - 4: Inhibit suspending the session or computer
+        * - 12: Inhibit the session being marked as idle
+        */
+        this.inhibitFlags= 12;
+        
+        // Caffeine state
         this._state = false;
         this._userState = false;
-        // who has requested the inhibition
-        this._last_app = '';
-        this._last_cookie = '';
-        this._apps = [];
-        this._cookies = [];
-        this._objects = [];
-
-        this.add_actor(this._icon);
-        this.add_style_class_name('panel-status-button');
-        this.connect('button-press-event', this.toggleState.bind(this));
-        this.connect('touch-event', this.toggleState.bind(this));
-
-        // Restore user state
-        if (this._settings.get_boolean(USER_ENABLED_KEY) && this._settings.get_boolean(RESTORE_KEY))
+        
+        // Store the inhibition requests until processed
+        this._inhibition_added_fifo=[];
+        this._inhibition_removed_fifo=[];
```

Comment:
To keep the style consistent:
```suggestion
        this._inhibitionAddedFifo = [];
        this._inhibitionRemovedFifo = [];
```